### PR TITLE
Feature/database

### PR DESCRIPTION
**Pull request by Oluwashenor**

### CODE REVIEW

File: src/Hng.Domain/Entities/Models/BaseModel.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace Hng.Domain.Entities.Models
+{
+    public class BaseModel
+    {
+        [Key]
+        public long Id { get; set; }
```
Comment (NdubuisiJr) [inline src/Hng.Domain/Entities/Models/BaseModel.cs:8]: Use Guid for the Id

File: src/Hng.Domain/Entities/Models/User.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿
+using Microsoft.AspNetCore.Identity;
+
+namespace Hng.Domain.Entities.Models
+{
+    public class User : IdentityUser<long>
```
Comment (NdubuisiJr) [inline src/Hng.Domain/Entities/Models/User.cs:6]: Don't do this. Don't use ASP.net identity service. Just create a plain user table. There are tickets for creating custom roles and claims and all that stuff. We don't need to hook up to ASP.net specific implementation.

File: src/Hng.Infrastructure/Context/MyDBContext.cs [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
-using System.Threading.Tasks;
+﻿using Hng.Domain.Entities.Models;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Identity.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore;
 
 namespace Hng.Infrastructure.Context
 {
-    public class MyDBContext : DbContext
+    public class MyDBContext(DbContextOptions options) : IdentityDbContext<IdentityUser<long>, IdentityRole<long>, long>(options)
```
Comment (NdubuisiJr) [inline src/Hng.Infrastructure/Context/MyDBContext.cs:8]: Use a plain DB context. Don't hook up to ASP.net Identity implementation

File: src/Hng.Infrastructure/Hng.Infrastructure.csproj [hunk 4]
```diff
@@ -7,6 +7,7 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Identity.EntityFrameworkCore" Version="8.0.6" />
```
Comment (NdubuisiJr) [inline src/Hng.Infrastructure/Hng.Infrastructure.csproj:10]: Remove all `Microsoft.AspNetCore.Identity.EntityFrameworkCore` reference as we are not using it.

File: src/Hng.Web/Hng.Web.csproj [hunk 5]
```diff
@@ -26,7 +26,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <ProjectReference Include="..\HNG-Boilerplate-Application\HNG-Boilerplate-Application.csproj" />
+    <!--<ProjectReference Include="..\HNG-Boilerplate-Application\HNG-Boilerplate-Application.csproj" />-->
```
Comment (NdubuisiJr) [inline src/Hng.Web/Hng.Web.csproj:29]: Why did you remove this reference?

File: src/Hng.Web/Program.cs [hunk 6]
```diff
@@ -9,11 +14,16 @@
 builder.Services.AddEndpointsApiExplorer();
 builder.Services.AddSwaggerGen();
 
-var connString = builder.Configuration.GetConnectionString("DefaultConnectionString");
+var connString = builder.Configuration.GetConnectionString("DefaultConnection");
```
Comment (NdubuisiJr) [inline src/Hng.Web/Program.cs:17]: Don't touch this. The Devops team are already using it

File: src/Hng.Web/Program.cs [hunk 7]
```diff
@@ -9,11 +14,16 @@
...
 builder.Services.AddConfiguredServices(connString);
 
 var app = builder.Build();
 
+using var scope = app.Services.CreateScope();
```
Comment (NdubuisiJr) [inline src/Hng.Web/Program.cs:22]: Move line 22 to 26 into an extension method.
Remove line 24. Create a generic repository instead.

**Review by NdubuisiJr — CHANGES_REQUESTED** [review]

**Comment by NdubuisiJr** [comment]
- Your branch has conflicts. Please update it with the latest changes in `dev` branch
- Create a Generic Repostory for accessing data for all entities.
- Create your models directly on the `Entities` folder

File: src/Hng.Web/Hng.Web.csproj (hunk 5 above)
Comment (Oluwashenor) [reply src/Hng.Web/Hng.Web.csproj:28]: The csproj file seems to be missing, so i had to comment it out

Comment (NdubuisiJr) [reply src/Hng.Web/Hng.Web.csproj:28]: Ok

File: src/Hng.Application/Services/Implementation/UserService.cs [hunk 8]
```diff
@@ -0,0 +1,29 @@
+﻿
+
+using Hng.Application.DTOs;
+using Hng.Application.Services.Interfaces;
+using Hng.Domain.Entities.Models;
+using Hng.Infrastructure.Context;
+
+namespace Hng.Application.Services.Implementation
+{
+    public class UserService(MyDBContext context) :IUserService
```
Comment (NdubuisiJr) [inline src/Hng.Application/Services/Implementation/UserService.cs:10]: Remove this service. You don't need it.

File: src/Hng.Web/ServiceExtensions/AppExtension.cs [hunk 9]
```diff
@@ -0,0 +1,20 @@
+﻿using Hng.Application.Services.Interfaces;
+using Hng.Infrastructure.Context;
+using Hng.Infrastructure.Seeder;
+using Microsoft.EntityFrameworkCore;
+
+namespace Hng.Web.ServiceExtensions
+{
+    public static class AppExtension
+    {
+        public static async Task<WebApplication> ExtendApp(this WebApplication app)
```
Comment (NdubuisiJr) [inline src/Hng.Web/ServiceExtensions/AppExtension.cs:10]: Call this method `UseSeeder`

**Review by NdubuisiJr — CHANGES_REQUESTED** [review]

**Comment by NdubuisiJr** [comment]
@Oluwashenor I like your implementation. You are almost there. Resolve the additional two comments above then:

- Create a [generic repository](https://learn.microsoft.com/en-us/aspnet/mvc/overview/older-versions/getting-started-with-ef-5-using-mvc-4/implementing-the-repository-and-unit-of-work-patterns-in-an-asp-net-mvc-application#create-a-generic-repository) in the infrastructure project
- Create a `UserService` in the infrastructure project using the `Generic Repository`
- Create a simple endpoint that uses the `UserService` to return the seeded users

The constructor of your `UserService` should look like this
```csharp
public UserService(IRepository<User> userRepository)
```
